_posts/2018-03-18-ar.markdown

```diff
@@ -0,0 +1,224 @@
+at release could only be used to find horizontal planes.
+
+Here's how our original approach worked:
+
+<article class="desktop-only" style='display: flex; flex-flow:row; margin: 0px -60px;'>
+<div style='flex:1;'>
+  <ol class="workflow" data-id="v1">
+    <li>Open up ARKit and let it discover the world</li>
+    <li>Detect the edge of a floor by asking a user to point at the floor near the wall</li>
+    <li>Find the edge furthest away</li>
+    <li>Attach a work to the floor closest to the wall</li>
+  </ul>
+</div>
+<div style='flex:1'>{% include svg/ar/v1.svg %}</div>
+</article>
+
+This could work, but it was hard to explain for a first time user. A user spent most of their time with their phone
+pointed at the floor, then we had no way to know if feature points had been detected all the way to the floor edges.
+
+From there, we moved on experimented with using the iPhone's location as a point of reference:
+
+<article class="desktop-only" style='display: flex; flex-flow:row; margin: 0px -60px;'>
+<div style='flex:1;'>
+  <ol class="workflow" data-id="v2">
+  <li>Open up ARKit and let it discover the world</li>
+  <li>Ask a user to put their phone against the wall</li>
+  <li>Detect that ARKit has disconnected, use the last known point as an anchor</li>
+  <li>When the user pulls back and reconnects, place the work</li>
+  </ul>
+</div>
+<div style='flex:1'>{% include svg/ar/v2.svg %}</div>
+</article>
+
+We got this to a point where it was working most of the time. One of the core issues is that their is a lot of
+guess-work at the end of the process. The Artwork would usually be at the position of your phone, but it was rarely
+oriented correctly. The positioning of the Artwork was tough in this case. We experimented by placing the artwork a
+meter away, but in user testing very few people could do that roughly in their head, and opted for the exact position of
+the phone - which meant that you had to touch the wall, then come back to see if it fit right.
+
+The user flow was hard to explain, and it felt like we weren't setting people up to win. It was quite a debate about
+whether we could ship with this or not. Then we read about the upcoming changes in ARKit 1.5. **Vertical plane
+detection**. Or, _mostly_ wall detection.
+
+With the advent of vertical wall detection, we could change our flow to be a lot like what someone would have thought it
+would be:
+
+<article class="desktop-only" style='display: flex; flex-flow:row; margin: 0px -60px;'>
+<div style='flex:1;'>
+  <ol class="workflow" data-id="v3">
+    <li>Open up ARKit and let it discover the world</li>
+    <li>Ask a user to point at a wall</li>
+    <li>Use the feature detection to detect a plane</li>
+    <li>Use that plane to place an Artwork</li>
+  </ul>
+</div>
+<div style='flex:1'>{% include svg/ar/v3.svg %}</div>
+</article>
+
+This is almost perfect, the only flaw is that this example does work in principal. You don't get enough feature points
```

Comment:
principle, not principal

_This seems almost perfect, except that it only works in theory._ In practice, ...